### ISSUE TITLE
Linking containers across hosts (--link)

### ISSUE THREAD

**Issue by tcolar** [zettio/weave#317]
Following up on #312 I was able to start my containers from the weave image on my remotes and that seem to be working properly.

Now I was using docker --link to create DNS-ish enrties for some containers that other containers need to connect to .... unsurprisingly --link didn't automatically work across hosts when using weave (trying to tcp access a container on host1 from a container on host2).

Is the only solution to that to use wavedns ? It work across hosts (shared DNS) ?

Thanks.

**Comment by rade** [comment]
> Is the only solution to that to use weavedns?

It's perhaps not the only solution but certainly what we'd recommend.

> It work across hosts (shared DNS) ?

Yes!

**Comment by tcolar** [comment]
OK, I suspected that's what it was intended for, I'll try it out, thanks.

**Comment by tcolar** [comment]
Sorry one more question, do only one "master" host need to do:
weave launch-dns 10.1.254.1/24

or do all hosts need that ?

**Comment by rade** [comment]
From [the docs](https://github.com/zettio/weave/tree/master/weavedns#weave-dns-server)

> It is run *per-host*

**Comment by tcolar** [comment]
So i was trying to test this, i endup with a convoluted call like this:

DOCKER_HOST=tcp://127.0.0.1:4001 docker run -v /var/run/docker.sock:/var/run/docker.sock -v /proc:/hostproc -v /tmp:/tmp --privileged=true --rm --env PROCFS=/hostproc --net=host igneoussystems/weave:latest 10.0.1.5/20 --name=Topo.slab0.lithos5942550.0 --privileged=true --with-dns --dns-search=weave.local lithos:dev /bin/sh -c ping -c 3 plume.weave.local

so i'm trying to run my container  lithos:dev via the weave container that is on my remote docker host.
The "/bin/sh -c ping -c 3 plume.weave.local bit" is juts temp stuff i added to test the dns stuff.

The problem is I get:

  flag provided but not defined: --with-dns
  Usage: docker run [OPTIONS] IMAGE [COMMAND] [ARG...]

In other words it seems the docker arg parsing is trying to consume the flags that should go to weave I think ....
 
Any suggestion ?

**Comment by rade** [comment]
The problem is not the wrapper container but the weave invocation itself. Weave is quite particular about where it wants its flags. From the usage:

> weave run        [--with-dns] <cidr> <docker run args> ...

So the `--with-dns` must come before the IP/prefix, whereas in your example it comes a long way after.

**Comment by tcolar** [comment]
ho I should have know that, I've messed enough with the flag package !

**Comment by rade** [comment]
Actually the reason weave is so particular about this is that `--with-dns` gets processed by the shell script and we have so far shied away from introducing full-blown option parsing in that.

**Comment by tcolar** [comment]
That's fair enough
